docs/cookbook/prompt_engineering/chaining_based/least_to_most.md

```diff
@@ -0,0 +1,142 @@
+# Least to Most: Enhancing LLM Reasoning with Subproblem Decomposition
+
+This recipe demonstrates how to implement the Least to Most technique using Large Language Models (LLMs) with Mirascope. Least to Most is a prompt engineering method that enhances an LLM's reasoning capabilities by breaking down complex problems into smaller, more manageable subproblems.
+
+??? tip "Mirascope Concepts Used"
+
+    - [Prompts](../../../learn/prompts.md)
+    - [Calls](../../../learn/calls.md)
+    - [Dynamic Configuration](../../../learn/dynamic_configuration.md)
+    - [Response Models](../../../learn/response_models.md)
+
+!!! note "Background"
+
+    [Least to Most](https://arxiv.org/pdf/2205.10625) is a more extensive version of [Chain of Thought](https://arxiv.org/abs/2201.11903), where separate calls are used to break down the original problem into subproblems as well as solve each individual step/subproblem. After solving each subproblem, the result is appended to the chat's history until the original problem is solved. Least to Most is an effective technique for symbolic and arithmetic reasoning tasks.
+
+## Implementing Least to Most
+
+Let's implement the Least to Most technique using Mirascope:
+
+```python
+from openai.types.chat import ChatCompletionMessageParam
+from pydantic import BaseModel, Field
+
+from mirascope.core import openai, prompt_template
+
+few_shot_examples = """
```

Comment:
again, this is somewhat tough to read, we should structure it and use list specifiers